assets/js/components/Dashboard/PermanentConfiguration/Workflows/GlEink/GlEinkWorkflow.tsx

```diff
@@ -66,24 +164,88 @@ const GlEinkWorkflow: ComponentType<WorkflowProps> = ({
         navigate(-1);
       };
       onBack = () => {
+        Object.keys(placesAndScreensToUpdate).forEach((placeId) => {
+          validationErrors[placeId] = [];
+        });
+        dispatch({
+          type: "SET_VALIDATION_ERRORS",
+          validationErrors: validationErrors,
+        });
         setConfigStep(configStep - 1);
       };
       onForward = () => {
-        putPendingScreens(
-          placesAndScreensToUpdate,
-          "gl_eink_v2",
-          configVersion
-        ).then(() => navigate("/pending"));
+        const fieldsWithErrors = validateRequiredFields(
+          placesAndScreensToUpdate
+        );
+
+        if (fieldsWithErrors.size === 0) {
+          putPendingScreens(
+            placesAndScreensToUpdate,
+            "gl_eink_v2",
+            configVersion
+          ).then((response) => {
+            if (response.ok) {
+              navigate("/pending");
+            } else {
+              return response
+                .json()
+                .then((data) => {
+                  validateDuplicateScreenIds(
+                    placesAndScreensToUpdate,
+                    data.duplicate_screen_ids
+                  );
+                  fieldsWithErrors.add("screen_id");
+                  setValidationErrorMessage(
+                    generateErrorMessage(fieldsWithErrors)
+                  );
+                  setShowValidationAlert(true);
+                  dispatch({
+                    type: "SET_VALIDATION_ERRORS",
+                    validationErrors: validationErrors,
+                  });
+                })
+                .catch((error) => {
+                  console.log(error);
+                });
+            }
```

Comment:
Can this be extracted into its own function? The nested anonymous functions are getting a little hard to follow.